### PR TITLE
Ml improvement

### PR DESCRIPTION
**Pull request by ZevHadid**
I already improved our classifier model using FastText models so it can better correlate between words. I also updated our model so now its a little more comprehensive. I deleted our old endpoint and created  two endpoints in the api.

## Summary by CodeRabbit

- **Chores**
	- Updated version control settings to ignore certain machine learning model files.

- **New Features**
	- Streamlined transaction prediction endpoints now provide a simplified response with a label and confidence score for both English and Indonesian inputs.
	- Introduced a new bilingual transaction classification model that leverages advanced neural network techniques, replacing a legacy implementation to offer improved prediction capabilities.
	- Added functionality for downloading and extracting FastText models, enhancing model training and inference processes.
	- Launched a FastAPI application for predicting transaction categories from input text.
	- Updated API documentation to include new health check endpoints and improved error handling descriptions.

### CODE REVIEW

**Review by NojinNojs — CHANGES_REQUESTED** [review]
image

Sepertinya masih ada error, tolong di fix terlebih dahulu

**Review by NojinNojs — CHANGES_REQUESTED** [review]
**Dear @ZevHadid** 

I appreciate your effort in submitting the pull request for ML improvement. I have reviewed the changes, and I noticed some issues in the implementation which require adjustments to ensure functionality and consistency.

### **Issues Identified:**
image
1. **Model Loading Error**: The `load_model` function references a file path (`transaction_classifier_model.keras`) that is not accessible. Kindly verify the existence of the file and ensure it is packaged properly as an accessible `.keras` zip file.
2. **oneDNN Warning Messages**: The logs indicate that oneDNN custom operations are enabled, which may lead to slight numerical discrepancies. Consider adding instructions or clarifications on how to disable these operations if needed. 
3. **Process Spawn Error**: There appears to be an error in the subprocess handling when starting `uvicorn` in the application. It might be beneficial to revisit the configuration in `moment-fastapi-app.py` or provide an alternative approach to ensure smooth execution

I recommend addressing these issues to avoid potential execution errors. Please let me know if you need any assistance or clarification.

Thank you for your contribution!

**Best regards, Aqsan**

File: machine-learning/moment-fastapi-app.py [hunk 1]
```diff
@@ -1,289 +1,68 @@
...
+
+# Endpoint
+@app.post("/predict-category")
+def predict_english(input: TextInput):
+    label, confidence = predict_transaction(input.text)
+    return {"language": "english", "label": label, "confidence": round(confidence, 4)}
+
+# For local testing
+if __name__ == "__main__":
+    uvicorn.run("moment-fastapi-app:app", host="0.0.0.0", port=8000, reload=True)
```
[the remark this answers is not in the document]
Comment (NojinNojs) [reply machine-learning/moment-fastapi-app.py]: Is this done now?

File: machine-learning/moment-fastapi-app.py [hunk 2]
```diff
@@ -1,289 +1,68 @@
...
-            }
-        },
-        "documentation": {
-            "openapi": "/docs",
-            "redoc": "/redoc"
-        }
-    }
+import os
+import uvicorn
+
```
[the remark this answers is not in the document]
Comment (NojinNojs) [reply machine-learning/moment-fastapi-app.py]: Is this done now?

File: machine-learning/moment-fastapi-app.py [hunk 3]
```diff
@@ -1,289 +1,68 @@
...
+        output = inputs * a
+        return tf.keras.backend.sum(output, axis=1)
+
+model = load_model(model_path, custom_objects={'AttentionLayer': AttentionLayer})
+
+with open(tokenizer_path, 'rb') as f:
+    tokenizer = pickle.load(f)
+
+with open(label_encoder_path, 'rb') as f:
+    label_encoder = pickle.load(f)
```
[the remark this answers is not in the document]
Comment (NojinNojs) [reply machine-learning/moment-fastapi-app.py]: Is this done now?
